packages/muelu/test/unit_tests_kokkos/CoalesceDropFactory_kokkos.cpp

```diff
@@ -283,6 +283,86 @@ TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL(CoalesceDropFactory_kokkos, DistanceLaplacianS
 
 }  // DistanceLaplacianScaledCut
 
+TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL(CoalesceDropFactory_kokkos, DistanceLaplacianSgndRugeStuebenDistribLump, Scalar, LocalOrdinal, GlobalOrdinal, Node) {
+#include <MueLu_UseShortNames.hpp>
+  typedef Teuchos::ScalarTraits<SC> STS;
+  typedef typename STS::magnitudeType real_type;
+  typedef Xpetra::MultiVector<real_type, LO, GO, NO> RealValuedMultiVector;
+  using TMT = Teuchos::ScalarTraits<typename Teuchos::ScalarTraits<Scalar>::magnitudeType>;
+
+  MUELU_TESTING_SET_OSTREAM;
+  MUELU_TESTING_LIMIT_SCOPE(Scalar, GlobalOrdinal, Node);
+  out << "version: " << MueLu::Version() << std::endl;
+
+  RCP<const Teuchos::Comm<int>> comm = Parameters::getDefaultComm();
+
+  Level fineLevel;
+  TestHelpers_kokkos::TestFactory<SC, LO, GO, NO>::createSingleLevelHierarchy(fineLevel);
+
+  RCP<Matrix> A = TestHelpers_kokkos::TestFactory<SC, LO, GO, NO>::Build1DPoisson(36);
+  fineLevel.Set("A", A);
+
+  Teuchos::ParameterList galeriList;
+  galeriList.set("nx", Teuchos::as<GlobalOrdinal>(36));
+  RCP<RealValuedMultiVector> coordinates = Galeri::Xpetra::Utils::CreateCartesianCoordinates<SC, LO, GO, Map, RealValuedMultiVector>("1D", A->getRowMap(), galeriList);
+
+  // Doctor coordinates with the goal that the filtered matrix drops all entries in of the lower triangular portion of the matrix (except for final row).
+
+  auto vals = coordinates->getDataNonConst(0);
+  double delta  = 1.1;
+  for (size_t i = 0; i < coordinates->getMap()->getLocalNumElements(); i++) vals[i] = pow(delta, coordinates->getMap()->getGlobalElement((LO)i)) / 35.;
```

Comment:
wrap this in `{..}` to make sure that `vals` gets released before anyone tries to access it on device.